File: src/main/java/codereview/simpleorder/repository/ItemRepository.java
package codereview.simpleorder.repository;

import codereview.simpleorder.domain.Item;
import jakarta.persistence.LockModeType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Lock;

import java.util.List;

public interface ItemRepository extends JpaRepository<Item, Long> {

    @Lock(value = LockModeType.OPTIMISTIC)
    List<Item> findByIdIn(List<Long> itemIds);
}

File: item/src/main/kotlin/io/philo/shop/item/repository/ItemRepository.kt
package io.philo.shop.item.repository

import io.philo.shop.item.domain.entity.Item
import org.springframework.data.jpa.repository.JpaRepository

interface ItemRepository : JpaRepository<Item, Long> {
    
    fun findByIdIn(itemIds: Collection<Long>): List<Item>
}